src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -91,6 +101,38 @@ export class UrlPanelContent extends Component<Props, State> {
     this.setUrl();
 
     window.addEventListener('hashchange', this.resetUrl, false);
+
+    if (this.props.anonymousAccess) {
+      (async () => {
+        const anonymousAccessParameters = await this.props.anonymousAccess!.getAccessURLParameters();
+
+        if (!this.mounted) {
+          return;
+        }
+
+        if (!anonymousAccessParameters) {
+          return;
+        }
+
+        this.setState({
```

Comment:
It does not matter if it is called once or twice, internally it is debounced and executed once if calls are close enough. Calling it twice is a bit cleaner from code perspective, IMHO. But I've combined both settings into a single call, since you were asking about this also in previous review.